### DISCUSSION TITLE
Creating a replica of a screenshot with Snipping tool

### DISCUSSION THREAD

**Discussion by Khuram-2521** [python-pillow/Pillow#7368] [Q&A] [answered]
I want to make a simple replica of an image taken in Audacity's macro dialog boxes. I have tried creating an image with Pillow that has the text and almost similar background which is apparently similar to the screenshot but not detectable by PyAutoGui on confidence range >= 0.7

Apparently there is an unordered color combination in the picture that involves audacity background and the text, I have tried changing ARGB values to reach 0.6 confidence but no success on pushing it forward. Any help would be really appreciated!

**Comment by radarhere** [comment]
Could you post a self-contained copy of the code that you are using, including an images that it uses as input?

**Reply by MasterOz786** [reply]
This account would carry on the discussion for the original poster from now on. Excuse me for the late reply, was surrounded by unfortunate events lately. 

The code is as follows:

```python
from PIL import Image, ImageFont, ImageDraw
def text_to_image(text: str, font_filepath: str) -> None:
    img = Image.new("RGBA", (125, 30), (240, 240, 240, 255))
    font = ImageFont.truetype(font_filepath, 20)
    draw = ImageDraw.Draw(img)

    draw.text((2, 2), text, font=font, fill=(45, 53, 63))    
    img.save('prevMacro.PNG')

text_to_image('  Macro_10', 'arial.ttf')
```
----

The image I want to match:
![AudacityScreenBackground](https://github.com/python-pillow/Pillow/assets/123957466/1cda1c8a-5d55-4cef-adfe-cf4c41e20f61)

The image I am able to create:
![prevMacro](https://github.com/python-pillow/Pillow/assets/123957466/520b1ca0-bc31-4058-bf6c-a049716eedd4)

Any help would be really appreciated!

**Reply by radarhere** [reply]
```python
from PIL import Image, ImageFont, ImageDraw

def text_to_image(text: str, font_filepath: str) -> None:
    img = Image.new("RGB", (107, 20), "#fff")
    font = ImageFont.truetype(font_filepath, 14)
    draw = ImageDraw.Draw(img)
    
    draw.line((2, 0, 2, 20), "#d0d1d6")
    draw.line((3, 0, 3, 20), "#92929a")

    draw.text((9, -1), text, font=font, fill=(45, 53, 63))
    img.save('prevMacro.PNG')

text_to_image('10 (Macro_10)', '/System/Library/Fonts/Supplemental/Arial.ttf')
```
This is closer.
![prevMacro](https://github.com/python-pillow/Pillow/assets/3112309/bac2a5db-9fc2-4f6e-9010-83194d3e0aa8)

**Reply by radarhere** [reply]
Was that close enough? If not, is there any simple way for us to be able to know when we do hit the mark?

**Reply by MasterOz786** [reply]
It was similar to the image text, but I think that the background of the Audacity screen is messing things up. The confidence level of detection obtained is still below the mark. Are there any possible way outs?

**Reply by radarhere** [reply]
I mean, you can open up the original image in Gimp and move parts of the image around until you just have the line and a speckled background

![background](https://github.com/python-pillow/Pillow/assets/3112309/aae3c3b0-b52c-40da-967e-751089b54100)

and then write text on top of it
```python
from PIL import Image, ImageFont, ImageDraw

def text_to_image(text: str, font_filepath: str) -> None:
    img = Image.open('background.png')
    font = ImageFont.truetype(font_filepath, 14)
    draw = ImageDraw.Draw(img)
    
    draw.text((9, -1), text, font=font, fill=(45, 53, 63))
    img.save('prevMacro.PNG')

text_to_image('10 (Macro_10)', '/System/Library/Fonts/Supplemental/Arial.ttf')
```
giving

![prevMacro](https://github.com/python-pillow/Pillow/assets/3112309/7af7b356-60f2-4319-a7a2-c7fd0f4a7e01)

but I then start to wonder about why you need to do this programmatically in the first place, since the perfect replica would just be the original image itself.

**Reply by radarhere** [reply]
Was that helpful?

**Comment by MasterOz786** [comment, marked as answer]
Sorry for the late reply, it was fixed by using tesseract instead.
